tools/embedded_data_analyzer.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/python3
+
+HELP_STRING = '''
+Tock embedded data analysis tool
+
+Generates a report of a Tock binary's embedded data usage in the form of a static HTML webpage.
+
+Prerequisite:
+    - The binary must be compiled for the RISC-V architecture.
+    - Turn off LTO. To do so, set lto to false in the top-level Cargo.toml file.
```

Comment:
The debug build does this already, can't we just use that?

Reply:
The user might want to use this tool on the release build to see what embedded data makes it into the final binary.

Reply:
Agreed, but once you disable LTO it's closer to a debug binary then a release binary anyway. Using a debug binary means it could work with make, so you can just do: `make html-analysis` and get results

Reply:
Yeah, the "debug binary" is equivalent to a release binary with LTO disabled

Reply:
I don't know which build the user might want to run the tool on, so I just printed a soft warning when LTO is turned on for either build. It will keep running and just give a warning to the user.

Reply:
Sounds like this should be a separate PR that can be done later.